Lib/fractions.py

```diff
@@ -115,15 +115,16 @@ def __new__(cls, numerator=0, denominator=None, *, _normalize=True):
                     raise ValueError('Invalid literal for Fraction: %r' %
                                      numerator)
                 numerator = int(m.group('num') or '0')
-                denom = m.group('denom')
-                if denom:
-                    denominator = int(denom)
+                denominator = m.group('den')
```

Comment:
I'd suggest reverting this set of changes; IIUC they're not needed for this PR, and I don't see any particular advantage to the rename. (In fact, I prefer the version where we don't have the same local name referring to both a string and an int.)

Reply:
> I don't see any particular advantage to the rename.

Naming of patterns seems to be more consistent in this case: num/den vs num/denom.

But you are right, this is not related to the PR.  I'll revert.

> I prefer the version where we don't have the same local name referring to both a string and an int

I don't thing there are.  We have den/denom for strings and denominator - which is an integer.